### PR TITLE
Home page components

### PR DESCRIPTION
**Pull request by tahmasious**

### CODE REVIEW

**Comment by tahmasious** [comment]
@amRashidi

File: MyMovie/src/components/browsepage/RowsOfPeopleCard.vue [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+</section>    
+</template>
+
+<script setup>
+import {API_BASE_URL} from '@/constants/api-constants'
+import PeopleCard from './PeopleCard.vue';
+import {useFetch} from '@/composable/useFetch.js'
+const {data ,isLoading, error, fetchWrapper} = useFetch(); 
+
+fetchWrapper(`${API_BASE_URL}3/person/popular`);
```
Comment (amRashidi) [inline MyMovie/src/components/browsepage/RowsOfPeopleCard.vue]: handle this env variable in fetch wrapper
also better to add this to mounted cycle

File: MyMovie/src/components/browsepage/TrendingSearchSingleItem.vue [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<template>
+    <router-link to="#"
```
Comment (amRashidi) [inline MyMovie/src/components/browsepage/TrendingSearchSingleItem.vue:2]: avoid evaluate to props with #

File: MyMovie/src/components/general/AddToWatchListBtn.vue [hunk 3]
```diff
@@ -13,14 +28,50 @@
...
+
+import {useFetch} from '@/composable/useFetch.js'
+import {API_BASE_URL} from '@/constants/api-constants'
+import { inject } from 'vue';
+
+const {data , isLoading, error,fetchWrapper} = useFetch([])
+
+const userData = inject('user');
+function addToWatchList() {
+    fetchWrapper(`${API_BASE_URL}3/account/${userData.value.id}/watchlist`, {'media_type': props.media_type, 'media_id': props.media_id, 'watchlist': true}, 'POST');
```
Comment (amRashidi) [inline MyMovie/src/components/general/AddToWatchListBtn.vue]: same for env variable

File: MyMovie/src/components/general/RowOfMainMovieCard.vue [hunk 4]
```diff
@@ -1,16 +1,14 @@
...
-          </div>
-        </section>
-      </section>
+    <section class="movie-poster-container mt-8">
+      <div class="flex justify-between mb-3 items-center">
+        <h2 class="text-white text-lg"><slot></slot></h2>
+      </div>
+      <div class="flex justify-between flex-wrap">
+      <div v-for="movie in movies" :key="movie.id">
+        <MainMovieCard :MovieID="movie.id" :genreID="movie.genre_ids[0]" :title="movie.title" :src="movie.poster_path" :rating="movie.vote_average" :releaseDate="movie.release_date"/>
```
Comment (amRashidi) [inline MyMovie/src/components/general/RowOfMainMovieCard.vue]: better to handle this in component
:item="movie"

File: MyMovie/src/components/general/RowsOfOptionedMovieCards.vue [hunk 5]
```diff
@@ -12,9 +12,10 @@
         <div class="flex flex-wrap justify-between">
         <div v-for="movie in movies" :key="movie.id">
             <MainMovieCard 
-            :genreID="movie.genre_ids[0]" :title="movie.title" 
+            :MovieID="movie.id"
```
Comment (amRashidi) [inline MyMovie/src/components/general/RowsOfOptionedMovieCards.vue]: same

File: MyMovie/src/components/homepage/ProfileContainer.vue [hunk 6]
```diff
@@ -0,0 +1,56 @@
+<template>
+    <div class="profile-container gap-5 hidden sm:inline-flex items-center">
+          <div class="inline notification">
+            <a href="#">
```
Comment (amRashidi) [inline MyMovie/src/components/homepage/ProfileContainer.vue:4]: replace all of this wirh router link

File: MyMovie/src/components/homepage/RowsOfTrendingMovies.vue [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+          <div class="flex justify-between mb-3 items-center">
+            <h2 class="text-white text-lg"><slot></slot></h2>
+          </div>
+          <div class="flex justify-between flex-wrap">
+            <TrendingMovieExpandCard 
+            v-for="movie in movies" :key="movie.id"
+            :title="movie.title" 
+            :backgroundImage="movie.backdrop_path" 
+            :imdbRate="movie.vote_average" 
+            :publishYear="new Date(movie.release_date).getFullYear()"
```
Comment (amRashidi) [inline MyMovie/src/components/homepage/RowsOfTrendingMovies.vue]: better to handle this in component

File: MyMovie/src/components/peoplepage/RowsOfCustomPeople.vue [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+        <div v-if="isLoading">
+            LOADING ...
+        </div>
+        <div v-else-if="error"></div>
+        <div v-else class="flex flex-wrap justify-between">
+            <PeopleCard v-for="person in listOfMovies" :key="person.id" 
+                :name="person.name"
+                :image-path="person.profile_path"
+                :known-credit="person.popularity"
+                :personID="person.id"
```
Comment (amRashidi) [inline MyMovie/src/components/peoplepage/RowsOfCustomPeople.vue]: handle this in componet

File: MyMovie/src/components/peoplepage/RowsOfCustomPeople.vue [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+import PrimaryBtn from '@/components/general/PrimaryBtn.vue';
+import {useFetch} from '@/composable/useFetch.js'
+import { ref, watch } from 'vue';
+const {data ,isLoading, error, fetchWrapper} = useFetch([]); 
+
+const page = ref(1)
+const listOfMovies = ref([])
+
+watch(page,async () => {
+    await fetchWrapper(`${API_BASE_URL}3/person/popular?page=${page.value}`);
```
Comment (amRashidi) [inline MyMovie/src/components/peoplepage/RowsOfCustomPeople.vue]: same for env variables

File: MyMovie/src/components/singlemoviepage/AddRating.vue [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+        </div>
+        <div v-else-if="data.success">
+            Rate submitted successfully
+        </div>
+        <div v-else class="flex-col items-center">
+            <p class="text-sm">How do you rate this movie ?</p>
+            <div class="star-box flex">
+                
+                <img
+                src="@/assets/rate.png"
```
Comment (amRashidi) [inline MyMovie/src/components/singlemoviepage/AddRating.vue:24]: use vfor

File: MyMovie/src/components/singlemoviepage/MediaSection.vue [hunk 11]
```diff
@@ -0,0 +1,100 @@
+<template>
+    <section class="media mb-6 w-full md:w-full">
+        <h2 class="text-secondary mb-2">MEDIA</h2>
+        <div
+        class="selector-container mb-3 text-secondary flex justify-start gap-4 text-md"
+        >
+            <span @click="activeTab = 'Backdrops'"
```
Comment (amRashidi) [inline MyMovie/src/components/singlemoviepage/MediaSection.vue]: use vfor

File: MyMovie/src/components/singlemoviepage/MediaSection.vue [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+            BaseURLWidth.value = `/w780`
+        }else if (activeTab.value == 'Logos'){
+            Images.value = data.value.logos;
+            BaseURLWidth.value = `/w185`
+        }else {
+            Images.value = data.value.posters;
+            BaseURLWidth.value = `/w185`
+        }
+    }
+}, {immediate : true})
```
Comment (amRashidi) [inline MyMovie/src/components/singlemoviepage/MediaSection.vue]: according to isloading i think there is no need for immidate

File: MyMovie/src/components/singlemoviepage/NotableCastSection.vue [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+
+<script setup>
+import SingleCast from './SingleCast.vue';
+import {useFetch} from '@/composable/useFetch.js'
+import { useRoute } from 'vue-router';
+import {API_BASE_URL} from '@/constants/api-constants'
+import { watch } from 'vue';
+const {data ,isLoading, error, fetchWrapper} = useFetch({'cast' : []})
+const route = useRoute(); 
+fetchWrapper(`${API_BASE_URL}3/movie/${route.params.id}/credits`)
```
Comment (amRashidi) [inline MyMovie/src/components/singlemoviepage/NotableCastSection.vue]: same for env variables

**Review by amRashidi — APPROVED** [review]

**Pull request merged**